modules/oe_whitelabel_agenda/oe_whitelabel_agenda.module

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @file
+ * OE Whitelabel Agenda module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\link_description\Plugin\Field\FieldType\LinkDescriptionItem;
+use Drupal\oe_agenda\Entity\DayInterface;
+use Drupal\oe_agenda\Entity\SessionInterface;
+
+/**
+ * Implements hook_preprocess_oe_agenda() for oe_default agenda.
+ */
+function oe_whitelabel_agenda_preprocess_oe_agenda__oe_default(array &$variables): void {
+  foreach (Element::children($variables['content']['oe_agenda_days']) as $index) {
+    $day = $variables['content']['oe_agenda_days'][$index]['#oe_agenda_day'];
+    assert($day instanceof DayInterface);
+    // Pass agenda days as accordion items to the template.
+    $variables['days'][] = [
+      'title' => $day->get('oe_day_date')->first()->view('default'),
+      'content' => $variables['content']['oe_agenda_days'][$index],
+    ];
+  }
+}
+
+/**
+ * Implements hook_preprocess_oe_agenda_session() for oe_default session.
+ */
+function oe_whitelabel_agenda_preprocess_oe_agenda_session__oe_default(array &$variables): void {
+  $session = $variables['entity'];
+  assert($session instanceof SessionInterface);
+  $renderer = \Drupal::service('renderer');
+  assert($renderer instanceof RendererInterface);
+
+  // Build items for description list pattern displaying session links.
+  $items = [];
+  foreach (['oe_session_venue', 'oe_session_online_link'] as $index => $field_name) {
+    $field = $session->get($field_name);
+    assert($field instanceof FieldItemListInterface);
+    if ($field->isEmpty() || !$field->access()) {
```

Comment:
We need to gather the access cache metadata and bubble it up here.

Reply:
This is done in 9cb8051